### PR TITLE
Initial API

### PR DESCRIPTION
**Pull request by RossIV**
First stab at the API with DRF

### CODE REVIEW

**Comment by kberzinch** [comment]
Does it make sense to put the API stuff in its own module?

File: onken/public/urls.py [hunk 1]
```diff
@@ -12,4 +22,6 @@
 
     path('login', django_cas_ng.views.login),
     path('logout', django_cas_ng.views.logout),
+
+    url(r'^api/', include(router.urls)),
```
Comment (kberzinch) [inline onken/public/urls.py]: Please use the `path` function here; [`url` will be deprecated soon](https://docs.djangoproject.com/en/2.0/ref/urls/#url).

File: onken/settings/development.py [hunk 2]
```diff
@@ -181,6 +183,19 @@
     'TAGS': {'env': 'dev'},
 }
 
+# DRF configuration
+REST_FRAMEWORK = {
+    # Use Django's standard `django.contrib.auth` permissions,
+    # or allow read-only access for unauthenticated users.
```
Comment (kberzinch) [inline onken/settings/development.py]: How do permissions work out of the box? Does unauthn have read-only? We don't want that.

File: onken/workspace/urls.py [hunk 3]
```diff
@@ -1,6 +1,24 @@
...
+    path('admin/login/', onken.workspace.views.login),
+    path('admin/logout/', django_cas_ng.views.logout),
+    path('admin/', admin.site.urls),
+
+    path('', onken.workspace.views.index, name="workspace_index"),
+
+    path('login', django_cas_ng.views.login),
+    path('logout', django_cas_ng.views.logout),
+
+    url(r'^api/', include(router.urls)),
```
Comment (kberzinch) [inline onken/workspace/urls.py]: See comment on public urls

File: requirements.txt [hunk 4]
```diff
@@ -5,3 +5,6 @@ django-tenants==2.0.0
 unittest-xml-reporting==2.2.0
 ddtrace==0.12.1
 raven==6.9.0
+djangorestframework
+markdown
+django-filter
```
Comment (kberzinch) [inline requirements.txt]: Please pin all to latest available on pypi.

Also, why are we requiring `markdown`?

File: onken/workspace/views.py [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
+@login_required
 def index(request):
     return HttpResponse("This is the workspace app. This workspace is %s." % request.tenant.name)
+
+
+def login(request):
+    if request.user.is_authenticated and not request.user.is_staff:
+        return HttpResponseForbidden()
+    else:
+        return django_cas_ng.views.login(request)
```
Comment (kberzinch) [inline onken/workspace/views.py]: Instead of redeclaring this view here, please re-use the one in the public module.

**Comment by kberzinch** [comment]
Also, can/should we reuse functions (views, serializers) across public/workspace apps instead of copy-pasting?

File: onken/workspace/urls.py [hunk 6]
```diff
@@ -1,6 +1,24 @@
...
+import django_cas_ng.views
+from django.contrib import admin
+
+router = routers.DefaultRouter()
+router.register(r'users', onken.workspace.api_views.UserViewSet)
+router.register(r'groups', onken.workspace.api_views.GroupViewSet)
 
 urlpatterns = [
-    path('', onken.workspace.views.index, name="workspace_index")
+    path('admin/login/', onken.workspace.views.login),
```
Comment (kberzinch) [inline onken/workspace/urls.py]: Per Slack, please remove admin paths.

File: onken/settings/development.py (hunk 2 above)
Comment (RossIV) [reply onken/settings/development.py]: Unauthenticated people can only see the API index, i.e. the list of endpoints. They cannot see any actual model content.

Comment (kberzinch) [reply onken/settings/development.py]: Ok, that's acceptable.

File: requirements.txt (hunk 4 above)
Comment (RossIV) [reply requirements.txt]: I _think_ that goes with DRF for the web-browsable half of the API.

File: onken/workspace/views.py (hunk 5 above)
Comment (RossIV) [reply onken/workspace/views.py]: Can you do that?

File: requirements.txt (hunk 4 above)
Comment (kberzinch) [reply requirements.txt]: The last two are optional per DRF docs. I'm okay keeping them, just wanted to double-check on why they were there.

Still pin the versions.

File: onken/workspace/views.py (hunk 5 above)
Comment (kberzinch) [reply onken/workspace/views.py]: actually I'm dumb this is for admin which is getting yanked anyway

Comment (kberzinch) [reply onken/workspace/views.py]: do remove this view though

**Comment by kberzinch** [comment]
> do we want to go ahead and yank admin back out of the workspace app?
<sub>[View in Slack](https://sccgt.slack.com/archives/CB8VAHW9Z/p1532660021000039)</sub>

File: onken/public/urls.py [hunk 7]
```diff
@@ -12,4 +21,6 @@
 
     path('login', django_cas_ng.views.login),
     path('logout', django_cas_ng.views.logout),
+
+    re_path(r'^api/', include(router.urls)),
```
Comment (kberzinch) [inline onken/public/urls.py]: Just `path`.

File: onken/workspace/urls.py [hunk 8]
```diff
@@ -1,6 +1,18 @@
...
+router.register(r'groups', onken.workspace.api_views.GroupViewSet)
 
 urlpatterns = [
-    path('', onken.workspace.views.index, name="workspace_index")
+    path('', onken.workspace.views.index, name="workspace_index"),
+
+    path('login', django_cas_ng.views.login),
+    path('logout', django_cas_ng.views.logout),
+
+    re_path(r'^api/', include(router.urls)),
```
Comment (kberzinch) [inline onken/workspace/urls.py]: Just `path`.

**Pull request merged**